File: 프로그래머스/2/42586. 기능개발/기능개발.swift
import Foundation

func solution(_ progresses:[Int], _ speeds:[Int]) -> [Int] {
    var days = [Int]()
    for (progress, speed) in zip(progresses, speeds) {
        var remain = 100 - progress
        var day = 0
        while remain > 0 {
            remain -= speed
            day += 1
        }
        days.append(day)
    }
    var cur = days[0]
    var cnt = 1
    var ret = [Int]()
    
    for i in 1..<days.count {
        if cur < days[i] {
            ret.append(cnt)
            cnt = 1
            cur = days[i]
        } else { cnt += 1 }
    }
    if cnt > 0 { ret.append(cnt) }
    return ret
}

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> days, ret;
    int n = speeds.size();
    for(int i = 0; i < n; i++) {
        int remain = 100 - progresses[i];
        int d = (remain + speeds[i] - 1) / speeds[i];
        days.push_back(d);
    }
    
    int mn = days[0];
    int cnt = 1;
    for(int i = 1; i < n; i++) {
        if(mn >= days[i]) cnt++;
        else {
            ret.push_back(cnt);
            cnt = 1;
            mn = days[i];
        }
    }
    ret.push_back(cnt);
    return ret;
}